### ISSUE TITLE
Improve accessibility

### ISSUE THREAD

**Issue by bglamadrid** [trebol-ecommerce/ngx-trebol-frontend#18]
_This issue is not meant to be assigned and be worked on its own. It is more akin to a call for action to raise relating issues, as described in the section "What needs to be done"._

## Context

I am absolutely NOT an expert in matters of web accessibility. I do know ARIA properties exist, but don't really understand when and where to use them. I've yet to learn that.

Still, as an open source software project, I do believe it is in the best interest of any and every potential collaborator, to make this as accessible as a web app can be.

## What has been done

- To follow appropriate naming & HTML structuring conventions, so that screen readers may not be too confused by the generated markup.
- To include recommended additional properties in the markup, e.g. `alt` in `img` elements to describe the content of each image.

You probably can already tell this isn't enough.

## What needs to be done

Discuss factors that need to be considered, corrected, or implemented in order to improve the web app's accessibility. For instance; add support for high contrast, dark and light modes. To disable animations. And other things.

Any of these factors can become issues of their own if someone is up for the task. But in order to do that, we should still have a better grasp on their concepts, preferably along with an illustrative example, or a reference; a webpage, a codepen, a well-written article, a technical blog post, or a through answer in Stack Overflow.

**Comment by ferrorenan** [comment]
Hey @bglamadrid  I can make some improvements like aria in the elements and some other things, that would be a start! What do you think?!

**Comment by bglamadrid** [comment]
Hi @ferrorenan that would be alright, but as I said in the description, first I'd like to acknowledge what things need to be done.
Because from what I've gathered at MDN:

> There is a saying "No ARIA is better than bad ARIA." In WebAim's survey of over one million home pages, they found that Home pages with ARIA present averaged 41% more detected errors than those without ARIA. While ARIA is designed to make web pages more accessible, if used incorrectly, it can do more harm than good.


This is why I consider this a `stub` issue. I would like that this is worked on, but first we should have a to-do list of things we can improve upon. Unless you yourself are willing to

- provide a reference document and
- contrast your every change against that reference

In short words, _We need to inspect the code first; find markup patterns that need improvement (and what is it that we're improving in each case); finally, improve markup with those recommendations._
